app/middleware/headerAndFooter.js

```diff
@@ -0,0 +1,48 @@
+const rp = require('request-promise-native');
+const NodeCache = require('node-cache');
+
+const headerFooterCache = new NodeCache();
+
+module.exports = async (_, res, next) => {
+  try {
+    // Check for cached data
+    const cachedData = headerFooterCache.get('nhsHeaderFooterApi');
+    if (cachedData) {
+      // Attach cached data to response
+      res.locals = {
+        ...res.locals,
+        footerLinks: cachedData.footerLinks,
+        headerLinks: cachedData.headerLinks,
+      };
+      return next();
+    }
+
+    // Get data from Wagtail endpoint
+    const data = await rp('https://www.nhs.uk/content-api/header-footer.json');
```

Comment:
Can the API URL be a variable so we can use integration, staging and live?

Reply:
This is still a WIP atm as we need to discuss with people how to get around the basic auth on lower environments

Reply:
Mike pointed me to these URLs which are not behind basic auth:
https://nhsuk-live.nhswebsite-integration.nhs.uk/content-api/header-footer.json
https://nhsuk-live.nhswebsite-staging.nhs.uk/content-api/header-footer.json

Reply:
Default of the integration URL has been added so it is easier for local development... I will add staging to Vault, infra task can be added as dependency of release for production env var to be added